ledger-core/virtual/integration/outgoing_test.go

```diff
@@ -266,3 +266,234 @@ func TestVirtual_CallMethodOutgoing_WithTwicePulseChange(t *testing.T) {
+
+		firstPulse  = server.GetPulse().PulseNumber
+		secondPulse pulse.Number
+		thirdPulse  pulse.Number
+
+		firstApprover  = gen.UniqueGlobalRef()
+		secondApprover = gen.UniqueGlobalRef()
+
+		firstVCallRequest, firstVDelegatedCallRequest = true, true
+
+		firstExpectedToken = payload.CallDelegationToken{
+			TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+			PulseNumber:       0, // fill later
+			Callee:            outgoing,
+			Outgoing:          outgoing,
+			DelegateTo:        server.JetCoordinatorMock.Me(),
+			Approver:          firstApprover,
+		}
+		secondExpectedToken = payload.CallDelegationToken{
+			TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+			PulseNumber:       0, // fill later
+			Callee:            outgoing,
+			Outgoing:          outgoing,
+			DelegateTo:        server.JetCoordinatorMock.Me(),
+			Approver:          secondApprover,
+		}
+
+		expectedVCallRequest = payload.VCallRequest{
+			CallType:         payload.CTMethod,
+			CallFlags:        payload.BuildCallFlags(barIsolation.Interference, barIsolation.State),
+			Callee:           objectBGlobal,
+			Caller:           outgoing,
+			CallSequence:     1,
+			CallSiteMethod:   "Bar",
+			CallReason:       outgoingCallRef,
+			CallRequestFlags: payload.BuildCallRequestFlags(payload.SendResultDefault, payload.RepeatedCall),
+			Arguments:        []byte("123"),
+		}
+	)
+
+	// add ExecutionMocks to runnerMock
+	{
+		outgoingCall := execution.NewRPCBuilder(outgoingCallRef, outgoing).CallMethod(objectBGlobal, classB, "Bar", []byte("123"))
+		objectAResult := requestresult.New([]byte("finish A.New"), outgoing)
+		objectAResult.SetActivate(reference.Global{}, classA, []byte("state A"))
+		objectAExecutionMock := runnerMock.AddExecutionMock("New")
+		objectAExecutionMock.AddStart(
+			func(_ execution.Context) {
+				server.IncrementPulseAndWaitIdle(ctx)
+				secondPulse = server.GetPulse().PulseNumber
+				firstExpectedToken.PulseNumber = secondPulse
+			},
+			&execution.Update{
+				Type:     execution.OutgoingCall,
+				Error:    nil,
+				Outgoing: outgoingCall,
+			},
+		)
+		objectAExecutionMock.AddContinue(
+			func(result []byte) {
+				require.Equal(t, []byte("finish B.Bar"), result)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: objectAResult,
+			},
+		)
+	}
+
+	// add checks to typedChecker
+	{
+		typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+			// check for pending counts must be in tests: call constructor/call terminal method
+			assert.Equal(t, outgoing, report.Object)
+			assert.Equal(t, payload.Empty, report.Status)
+			assert.Zero(t, report.DelegationSpec)
+			return false
+		})
+		typedChecker.VDelegatedCallRequest.Set(func(request *payload.VDelegatedCallRequest) bool {
+			assert.Equal(t, outgoing, request.Callee)
+
+			msg := payload.VDelegatedCallResponse{
+				Callee: request.Callee,
+				ResponseDelegationSpec: payload.CallDelegationToken{
```

Comment:
here you are actually building tokens. I think you can fill firstExpectedToken/secondExpectedToken here with all the fields.

Reply:
I just want to use different variables to be sure that they don't affect each other.

Reply:
well, system logic is - token that is provided by VDelegatedCallResponse should be used in messages. So they are supposed to be the same. And creating them here and sending them back to node should not be a problem (given that VDelegatedCallRequest/Response have tests)

Reply:
done